core/src/main/java/hudson/logging/LogRecorderManager.java

```diff
@@ -119,8 +150,8 @@ public HttpResponse doNewLogRecorder(@QueryParameter String name) {
         Jenkins.get().checkPermission(Jenkins.ADMINISTER);
 
         Jenkins.checkGoodName(name);
-        
-        logRecorders.put(name,new LogRecorder(name));
+
+        recorders.add(new LogRecorder(name));
```

Comment:
Do we not have to sync the legacy field here?

Reply:
generally this is just to maintain API compatibility, if this does or doesn't it won't affect users.

I plan to file PRs to update consumers to the new API anyway.